archive/extract_test.go

```diff
@@ -142,19 +145,25 @@ func testExtract(t *testing.T, when spec.G, it spec.S) {
 		})
 
 		it("fails if file exists where directory needs to be created", func() {
+			oldUmask := archive.SetUmask(0)
+			defer archive.SetUmask(oldUmask)
+
 			file, err := os.Create(filepath.Join(tmpDir, "root"))
 			h.AssertNil(t, err)
 			h.AssertNil(t, file.Close())
 
-			h.AssertError(t, archive.Extract(tr), "failed to create directory")
+			h.AssertError(t, archive.Extract(tr, 0), "failed to create directory")
 		})
 
 		it("doesn't alter permissions of existing folders", func() {
+			oldUmask := archive.SetUmask(0)
```

Comment:
Could we have a test that asserts the permissions on the directory created using `oldUmask`?

Reply:
I think in theory that is what this test is supposed to show. However it doesn't seem to make any difference what value is passed in...

Reply:
Added a test!